### PR TITLE
Fix #311 replace cytoscape with react force graph

### LINKED ISSUE #311: Use force graph to display graph

### PR DESCRIPTION
**Pull request by Anchel123**
## Release Notes

- **New Features**
  - Introduced a new `GraphView` component for rendering 2D force-directed graphs.
  - Added dynamic fetching of options in the `Combobox` component based on user interaction.
  - Enhanced the `Chat` component to support new data structures for paths.

- **Bug Fixes**
  - Improved error handling in the `Combobox` when fetching options.

- **Updates**
  - Updated the `DataPanel` to exclude additional properties from display.
  - Modified the `Toolbar` component to simplify state management and functionality.

- **Tests**
  - Expanded test coverage for chat functionality and the `CodeGraph` component, including new scenarios for error handling and UI interactions.

- **Dependencies**
  - Updated several package versions to ensure compatibility and performance improvements.

### CODE REVIEW

File: app/components/chat.tsx [hunk 1]
```diff
@@ -151,90 +147,65 @@ export function Chat({ repo, path, setPath, graph, chartRef, selectedPathId, isP
     }, [path])
 
     useEffect(() => {
-        if (isPath) return
+        if (isPathResponse || isPathResponse === undefined) return
```
Comment (AviAvni) [inline app/components/chat.tsx:150]: you can just leave it like
if (isPathResponse) return

File: app/components/chat.tsx [hunk 2]
```diff
@@ -151,90 +147,65 @@ export function Chat({ repo, path, setPath, graph, chartRef, selectedPathId, isP
...
-                            e.style(PATH_EDGE_STYLE)
+                        if (prev.links.some(e => e.id === id) && !p.links.some(e => e.id === id)) {
+                            link.isPathSelected = false
                         }
                     })
                 } else {
-                    const elements = chart.elements().filter(e => [...prev.edges, ...prev.nodes].some(el => el.id == e.id() && ![...p.nodes, ...p.edges].some(ele => ele.id == e.id()))).removeStyle()
-                    if (isPathResponse) {
+                    const elements = graph.getElements().filter(e => [...prev.links, ...prev.nodes].some(el => el.id === e.id && ![...p.nodes, ...p.links].some(ele => ele.id === el.id)))
+                    if (isPathResponse || isPathResponse === undefined) {
```
Comment (AviAvni) [inline app/components/chat.tsx:169]: same as prev comment

File: app/components/chat.tsx [hunk 3]
```diff
@@ -369,7 +312,7 @@ export function Chat({ repo, path, setPath, graph, chartRef, selectedPathId, isP
                     ])
 
                     if (isPathResponse) {
-                        chartRef.current?.elements().removeStyle().layout(LAYOUT).run()
+                        // chartRef.current?.elements().removeStyle().layout(LAYOUT).run()
```
Comment (AviAvni) [inline app/components/chat.tsx:315]: delete

File: package.json [hunk 4]
```diff
@@ -39,14 +34,11 @@
...
     "@playwright/test": "^1.49.1",
-    "@types/cytoscape": "^3.21.8",
-    "@types/cytoscape-fcose": "^2.2.4",
-    "@types/node": "^22.10.2",
     "@types/react": "^18",
-    "@types/react-cytoscapejs": "^1.2.5",
     "@types/react-dom": "^18",
     "@types/react-syntax-highlighter": "^15.5.13",
     "eslint": "^9",
+    "react-force-graph-2d": "^1.25.8",
```
Comment (AviAvni) [inline package.json:41]: why this is on dev dependency?

**Review by AviAvni — CHANGES_REQUESTED** [review]

File: app/components/chat.tsx (hunk 1 above)
Comment (Anchel123) [reply app/components/chat.tsx:150]: no it says if isPathResponse is true or undefined then return

Comment (Anchel123) [reply app/components/chat.tsx:150]: undefined is false

**Comment by Naseem77** [comment]
This pull request closes issues: #333, #334, #335, and #336

**Review by AviAvni — APPROVED** [review]

**Pull request merged**